### DISCUSSION TITLE
2023-06-28 Demo of CUE + Terraform + GitHub experiment (followed by Q&A)

### DISCUSSION THREAD

**Discussion by myitcv** [cue-lang/cue#2462] [Show and tell]
As shared on the [2023-06-20 community call](https://github.com/cue-lang/cue/discussions/2441), we have been experimenting  using CUE to drive Terraform to configure our GitHub organisations and repositories.

@jpluscplusm and @myitcv will share progress from this experiment, and would love to hear your thoughts and feedback!

The session will be on Wednesday, **28 June, 2023 at 1530 UTC** and will run for about 1 hour.

We will begin with a demo of the running system, which will be followed by Q&A.

Please join us!

### Meeting details

https://meet.google.com/ncn-fbka-yrb

This session will also be recorded for those unable to join us live.

### CUE Community shared calendar

[Subscribe to the shared calendar](https://calendar.google.com/calendar/u/0?cid=Y19lNzkxMWQ5OWQ4ZGIyMmU2ZTVjMzhkMTVkNjY2ZTVlNjdiNWE5ODNkZWU4N2JmNTU2NDY3NzI1OGIxYjJhMTFhQGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20) for this and other events.

### Recording

https://youtu.be/ZT7XcRqekNw

### Slides/materials

https://docs.google.com/presentation/d/1DdUywroZ5N4fZY6eFW7hBesSQA6SGUIK9QkAe7b8ak4/edit?usp=sharing

### Auto-generated transcript 🆕 

https://docs.google.com/document/d/1Cwa426cw2C9WXBSzXsZ7SUsNsIn8ZOlhMSqOQRnovo8/edit?usp=sharing

### Example repository

https://github.com/cue-examples/cue-terraform-github-config-experiment

**Comment by aranw** [comment]
Hey @myitcv I unfortunately missed the call yesterday but was wondering whether you have the link to the recording as am keen to see CUE + Terraform

**Comment by myitcv** [comment]
I've just updated the discussion with a link to the recording, slides and auto-generated transcript (which makes for some fun reading!)

cc @aranw

**Comment by akselleirv** [comment]
Hello @myitcv, do you have a link to the repository?

**Reply by jpluscplusm** [reply]
It's not public just yet, but will be real soon now!

**Reply by myitcv** [reply]
Just added to the description.

**Comment by myitcv** [comment]
I've just updated the original discussion description with a link to the example repo. 

Please be reminded that this is just an experiment! Use at your own risk etc 😄

**Comment by joebuilds** [comment]
Why use make instead of `cue cmd`?

**Reply by myitcv** [reply]
This experiment largely focuses on the combination of CUE + Terraform, as opposed to the use of `cue cmd`. Hence the choice of `make` over `cue cmd` for the "task" aspects of the experiment largely reflect the tooling preference of @jpluscplusm. 

That said, we have use `cue cmd` tasks in a couple of places.

And not only that, I think (as discussed on the [demo](https://youtu.be/ZT7XcRqekNw)) there is potentially a much larger role for [`tools/flow`](https://pkg.go.dev/cuelang.org/go/tools/flow) (`cue cmd` depends on this package) in this space, to more clearly describe and reflect dependencies.

**Comment by infogulch** [comment]
It's a year later now, how has this worked out for you guys? Any thoughts or learnings since the presentation?

**Reply by hansbogert** [reply]
Been 2y now.  Would like to know as well.
